services/github/github-actions-workflow-status.service.js

```diff
@@ -0,0 +1,101 @@
+import Joi from 'joi'
+import { isBuildStatus, renderBuildStatusBadge } from '../build-status.js'
+import { NotFound } from '../index.js'
+import { GithubAuthV3Service } from './github-auth-service.js'
+import { documentation, errorMessagesFor } from './github-helpers.js'
+
+const schema = Joi.object({
+  workflow_runs: Joi.array()
+    .items(
+      Joi.object({
+        conclusion: Joi.alternatives()
+          .try(isBuildStatus, Joi.equal('no status'))
+          .required(),
+      })
+    )
+    .required()
+    .min(0)
+    .max(1),
+}).required()
+
+const queryParamSchema = Joi.object({
+  event: Joi.string(),
+  branch: Joi.string().required(),
```

Comment:
I've made branch a required param for this badge. Not my favourite solution, but workflow can contain a `/` and `branch=HEAD` doesn't work so we can't make the default "whatever the default branch is" in one request, so required query param it is. Lets lay that marker down while we're making a breaking change.

I don't think "latest workflow status across all branches" is a sensible default.

Reply:
Do you know off hand if this has to explicitly be a branch or will the upstream endpoint accept any valid ref?

Reply:
The docs say

> `branch` string
>
> Returns workflow runs associated with a branch. Use the name of the branch of the `push`.

Anecdotally, I tried some calls with a tag and a commit hash (which I expected would return a valid result if this could accept any ref). They returned 0 results.